cep-20.2.md

```diff
@@ -0,0 +1,832 @@
+  format
+
+## Major differences with conda-build
+
+Outputs and top-level package are mutually exclusive, and outputs have exactly the same structure as top-level keys.
+If outputs exist, the top-level keys are 'merged' with the output keys (e.g. for the about section).
+
+# Format
+
+## Schema version
+
+The implicit version of the YAML schema for a recipe is an integer 1.
+To discern between the "old" format and the new format, we utilize the file name.
+The old format is `meta.yaml` and the new format is `recipe.yaml`.
+The version can be explicitly set by adding a `schema_version` key to the recipe.
+
+```yaml
+# optional, since implicitly defaults to 1
+schema_version: 1 # integer
+```
+
+To benefit from autocompletion, and other LSP features in editors, we can add a schema URL to the recipe.
+
+```yaml
+# yaml-language-server: $schema=https://raw.githubusercontent.com/prefix-dev/recipe-format/main/schema.json
+```
+
+## Context section
+
+The context section is a dictionary with key-value pairs that can be used for string interpolation.
+The right-hand side of the key-value pair is a scalar (bool, number or string).
+
+The variables can reference other variables from the context section.
+
+> ![NOTE]
+> The order of the keys is not enforced by the YAML standard. We expect parsers to parse maps (especially the context section)
+> in the order they are defined in the file. However, we do not require this behavior for the recipe to be valid to conform to the YAML standard.
+> Given this, implementations need to ensure topological sorting is done before string interpolation.
+
+```yaml
+# The context dictionary defines arbitrary key-value pairs for Jinja interpolation
+# and replaces the {% set ... %} commands commonly used in recipes
+context:
+  variable: test
+  # note that we can reference previous values, that means that they are rendered in order
+  other_variable: test_${{ variable }}
+```
+
+
+
+## Package section
+
+```yaml
+package:
+  # The name of the package
+  name: string
+  # The version of the package, following the conda version spec
+  # Note that versions are always strings (1.23 is _not_ a valid version since it is a `float`. Needs to be quoted.)
+  version: string
+```
+
+## Build section
+
+```yaml
+build:
+  # the build number
+  number: integer  # defaults to 0
+
+  # the build string. This is usually ommited (can use `${{ hash }}` variable here)
+  string: string  # defaults to a build string made from "package hash & build number"
+
+  # A list of Jinja conditions under which to skip the build of this package (they are joined by `or`)
+  # What is valid in a an `if:` condition is valid
+  skip: [list of expressions]
+
+  # wether the package is a noarch package, and if yes, wether it is "generic" or "python"
+  noarch: OneOf<"generic" | "python">
+
+  # script can be a single string or a list of strings
+  # if script is a single string and ends with `.sh` or `.bat`, then we interpret it as a file
```

Comment:
```suggestion
  # if script is a single-line string and ends with `.sh` or `.bat`, then we interpret it as a file
```

I guess? :D

Reply:
Wait, we also have a `script` section below. Isn't this redundant? I'd rather just keep the section and forget about this key in `build`.